compiler/src/dotty/tools/dotc/transform/PatternMatcher.scala

```diff
@@ -326,7 +326,12 @@ object PatternMatcher {
           sym.isAllOf(SyntheticCase) && sym.owner.is(Scala2x)
 
         if (isSyntheticScala2Unapply(unapp.symbol) && caseAccessors.length == args.length)
-          matchArgsPlan(caseAccessors.map(ref(scrutinee).select(_)), args, onSuccess)
+          def tupleSel(sym: Symbol) = ref(scrutinee).select(sym)
+          def tupleApp(i: Int) = // manually inlining the call to NonEmptyTuple#apply, because it's an inline method
+            ref(defn.RuntimeTuplesModule).select(defn.RuntimeTuples_apply).appliedTo(ref(scrutinee), Literal(Constant(i)))
+          val isGenericTuple = tree.tpe.derivesFrom(defn.PairClass)
```

Comment:
Actually all `TupleN` derive from `Pair`, this dependency is injected by the compiler. You will probably need 
```scala
val isGenericTuple = !defn.isTupleNType(tree.tpe/*.widenDealias*/)
```